### PR TITLE
Refactors the SPIFFS flash driver for the F7 into a generic stm32 flash driver.

### PR DESCRIPTION
**Pull request by balazsracz**
This generic flash driver can be used with FlashFile<Stm32Flash> to create simpler storage systems.
Adds a specific list of flash sector offsets to correctly support the F7 flash.

### CODE REVIEW

File: src/freertos_drivers/st/Stm32Flash.hxx [hunk 1]
```diff
@@ -24,46 +24,125 @@
...
+
 #include "stm32f_hal_conf.hxx"
 
-//
-// Stm32SPIFFS::flash_read()
-//
-int32_t Stm32SPIFFS::flash_read(uint32_t addr, uint32_t size, uint8_t *dst)
+/// Strategy module for finding flash erase sector numbers when the sectors are
+/// constant size.
+template <uint32_t ERASE_PAGE_SIZE> struct FlashFixedSectors {
```
Comment (bakerstu) [inline src/freertos_drivers/st/Stm32Flash.hxx:44]: Place brace on a new line.

File: src/freertos_drivers/st/Stm32Flash.cxx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * This file implements a generic Flash driver specific to STM32 F7 flash
+ * API.
+ *
+ * @author Balazs Racz
+ * @date 4 Sep 2023
+ */
+
+#include "freertos_drivers/st/Stm32Flash.hxx"
+
+uint32_t STM32F7_DUAL_BANK_2M_FLASH[] = {
```
Comment (bakerstu) [inline src/freertos_drivers/st/Stm32Flash.cxx:38]: Place brace on a new line. Below too.

**Review by bakerstu — APPROVED** [review]

File: src/freertos_drivers/st/Stm32Flash.hxx (hunk 1 above)
Comment (balazsracz) [reply src/freertos_drivers/st/Stm32Flash.hxx]: done

File: src/freertos_drivers/st/Stm32Flash.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/st/Stm32Flash.cxx]: done

**Pull request merged**